### PR TITLE
[fix](array-type) revert ARRAY<NOT_NULL(T)>

### PR DESCRIPTION
**Pull request by cambyzju**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

Now there are still lots of bugs related to ARRAY<NOT_NULL(T)>.
We decide not to support nested elements NOT NULLABLE for ARRAY type at the first version.
Nested elements of ARRAY will be always nullable.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [x] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by xy720 — COMMENTED** [review]
You should also update the return type of FunctionCallExpr
In some case, the array function's return type have `ContainsNull = false`.

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 1]
```diff
@@ -4847,8 +4847,6 @@ type ::=
   {: RESULT = ScalarType.createVarcharType(-1); :}
   | KW_ARRAY LESSTHAN type:value_type GREATERTHAN
   {: RESULT = new ArrayType(value_type); :}
-  | KW_ARRAY LESSTHAN KW_NOT_NULL LPAREN type:value_type RPAREN GREATERTHAN
-  {: RESULT = new ArrayType(value_type, false); :}
```
Comment (adonis0147) [inline fe/fe-core/src/main/cup/sql_parser.cup:4851]: I think we can keep this syntax but just throw an exception due to we don't support it temporarily.

**Review by adonis0147 — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/cup/sql_parser.cup (hunk 1 above)
Comment (cambyzju) [reply fe/fe-core/src/main/cup/sql_parser.cup:4851]: Lots of codes who called `new ArrayType` will also need to catch the exception,  if we throw an exception while new ArrayType. I think just remove these too lines is better.

**Review by xy720 — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 2]
```diff
@@ -4847,8 +4847,6 @@ type ::=
   {: RESULT = ScalarType.createVarcharType(-1); :}
   | KW_ARRAY LESSTHAN type:value_type GREATERTHAN
   {: RESULT = new ArrayType(value_type); :}
```
Comment (adonis0147) [inline fe/fe-core/src/main/cup/sql_parser.cup:4849]: What about using the following modification?

```suggestion
  {:
      RESULT = new ArrayType(value_type);
      if (!((ArrayType) RESULT).getContainsNull()) {
          throw new AnalysisException("ARRAY<NOT_NULL()> is unsupported.")
      }
  :}
```

**Review by adonis0147 — CHANGES_REQUESTED** [review]

**Review by adonis0147 — APPROVED** [review]
LGTM

**Pull request merged**